pkg/client/fake/client_test.go

```diff
@@ -1472,6 +1479,36 @@ var _ = Describe("Fake client", func() {
 		objOriginal.Status.NodeInfo.MachineID = "machine-id-from-status-update"
 		Expect(cmp.Diff(objOriginal, actual)).To(BeEmpty())
 	})
+	It("should not overwrite status fields of typed objects that have a status subresource on status update", func() {
```

Comment:
I think the tittle is incorrect and the number of negations makes me 🤯 . Did you mean to say `should not overwrite non-status fields of typed objects that have a status subresource on status update`? If yes, maybe express it as `Should only override status fields of typed objects that have a status subresource on status update`?

Reply:
I was trying to capture the impact that doing the workaround (which doesn't exist anymore) would only update the status fields that we bring to the update and not overwrite the status fields that exists already with blank fields, etc.  _(still a useful test)_

Changing it to what you suggested provides a clearer understanding though of what the test provides.  Negating the negations always  🤯

The inference of the update should include those fields not changing but asserting them in this test demonstrates that update only touching the fields it needs to write to.